### PR TITLE
Add Python 3.6 support.

### PR DESCRIPTION
**Pull request by lukesneeringer**
This PR adds CI testing for Python 3.6 (now possible since grpc has wheels).
It also updates the grpc lower bound to 1.1.3, since that is the oldest version with wheels available.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +11,11 @@ matrix:
   # Disabled until gRPC supports PyPy.
   # - python: pypy
   #   env: TOXENV=pypy
-  - python: 3.5
+  - python: 3.6
```
Comment (jonparrott) [inline .travis.yml]: We ran into some trouble with pylint on 3.6, so we held this env back to 3.5. You might need to do the same.

File: setup.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
     'dill>=0.2.5, <0.3dev',
     'future>=0.16.0, <0.17dev',
     'googleapis-common-protos>=1.5.2, <2.0dev',
-    'grpcio>=1.0.0, <2.0dev',
+    'grpcio>=1.1.3, <2.0dev',
```
Comment (jonparrott) [inline setup.py]: I'd personally leave this at `1.0.0` or `1.1.0`, as this *works* on those version, you'll just have to build the wheel yourself.

File: .travis.yml (hunk 1 above)
Comment (lukesneeringer) [inline .travis.yml]: What was the problem you had?

File: setup.py (hunk 2 above)
Comment (lukesneeringer) [inline setup.py]: Any idea where that will cause problems in the real world outside of CI? (My guess is not,  but I was unsure. I guess if the user _already has_ grpcio installed, obviously it is not a problem...)

File: .travis.yml (hunk 1 above)
Comment (jonparrott) [inline .travis.yml]: pylint didn't support 3.6, so it was failing outright.

File: setup.py (hunk 2 above)
Comment (jonparrott) [inline setup.py]: I'm more worried if downstream users pin a broader range.

Comment (lukesneeringer) [inline setup.py]: I buy it, but bumping to 1.0.2 to match GAPICs.

File: .travis.yml (hunk 1 above)
Comment (lukesneeringer) [inline .travis.yml]: Hmm. Working for me. Assuming CI passes again, will merge.

Comment (jonparrott) [inline .travis.yml]: SG.

**Pull request merged**